exact_string_matching/backward.py

```diff
@@ -136,3 +136,28 @@ def check_ends(i, k):
       j = 1
     skip[i + m - 1] = m - j
     i = i + BM[j]
+
+def bitap_shift_add(text: str, word: str, n: int, m: int):
+    Sigma = set()
```

Comment:
Nit: I used a convention in the code than A denotes the alphabet - could you change the naming?